apps/web/pages/event-types/[type]/index.tsx

```diff
@@ -563,6 +564,20 @@ const EventTypePage = (props: EventTypeSetupProps) => {
           form={formMethods}
           id="event-type-form"
           handleSubmit={async (values) => {
+            const getDirtyFields = (): Partial<FormValues> => {
+              const dirtyFields: Partial<{ [K in keyof FormValues]: FormValues[K] }> = {};
+              const watchedFields = values;
+
+              // Iterate over watched fields and compare with initial values
+              Object.keys(watchedFields).forEach((key) => {
+                const typedKey = key as keyof FormValues;
+                if (!isEqual(watchedFields[typedKey], defaultValues[typedKey])) {
+                  Object.assign(dirtyFields, { [typedKey]: watchedFields[typedKey] });
+                }
+              });
+              return dirtyFields;
+            };
+            const dirtyFields = getDirtyFields();
```

Comment:
Instead of doing this deep comparision we can actually leverage react hook form a bit more to get this 

-> Import `useFormState` from "react-hook-form"
-> We can grab the dirty fields as follows

  const { dirtyFields: dirtyFieldsFromHook } = useFormState({ control: formMethods.control });
  
  This will return a  `{ [name]:boolean }` object with all fields that have been touched. We should iterate over that and grab the values. 
  
 This is much cleaner. 
 
 I also noticed the method we have created here is not 100% correct and returns fields that i've not touched when just changing title.

Reply:
Yeah I agree. The dirtyFields wasn't working as expected so I tried a workaround. Needless to say, it's way better to amke dirtyFields work instead, and that's what @emrysal recommends as well. That's the move ahead as well. 🙏

Reply:
@alishaz-polymath @sean-brydon  so I saw the exact same issue of the dirtyFields not working as expected for event-types. [RHF docs](https://www.react-hook-form.com/api/useform/formstate/) state this `Important: Make sure to provide defaultValues at the useForm, so hook form can have a single source of truth to compare each field's dirtiness.` Is it possible that we've missed providing some default values in the useForm inputs for event types and that is causing this weird behaviour for the dirtyFields value?

Reply:
That was the first thing I checked as indeed that was the first suspect. As it turns it out the cause is more complicated than this.

Reply:
@alishaz-polymath can you try to use the RHF `dirtyFields` again after PR #13440 gets merged? The default value for the destinationCalendar was not set previously which has been fixed in that PR so hopefully `dirtyFields` should work as expected now

Reply:
There are other fields I'm concerned about, not in particular destinationCalendar. Fields such as `bookingFieds` for example is populated in a different way and as such doesn't recognize changes made when adding a question, but when removing one. So, it's not directly related to defaultValues. This is one example, but there are other fields in a similar pot. 🙏